components/nimbus/ios/Nimbus/Nimbus.swift

```diff
@@ -49,23 +47,68 @@ private extension Nimbus {
 }
 
 // Glean integration
-private extension Nimbus {
-    func recordExposure(experimentId _: String) {
-        // TODO: https://jira.mozilla.com/browse/SDK-209
+internal extension Nimbus {
+    func recordExposure(experimentId: String) {
+        let activeExperiments = getActiveExperiments()
+        if let experiment = activeExperiments.first(where: { $0.slug == experimentId }) {
+            GleanMetrics.NimbusEvents.exposure.record(extra: [
+                .experiment: experiment.slug,
+                .branch: experiment.branchSlug,
+                .enrollmentId: experiment.enrollmentId,
+            ])
+        }
     }
 
     func postEnrollmentCalculation(_ events: [EnrollmentChangeEvent]?) {
-        guard events?.isEmpty == false else {
-            return
-        }
-
-        // TODO: https://jira.mozilla.com/browse/SDK-209
+        // We need to update the experiment enrollment annotations in Glean
+        // regardless of whether we recieved any events. Calling the
+        // `setExperimentActive` function multiple times with the same
+        // experiment id is safe so nothing bad should happen in case we do.
         let experiments = getActiveExperiments()
-        experiments.forEach { experiment in
-            Glean.shared.setExperimentActive(experimentId: experiment.slug, branch: experiment.branchSlug, extra: nil)
+        recordExperimentTelemetry(experiments)
+
```

Comment:
Suggest we add an empty check for events after recording experiment enrollment.

```swift
guard let events = events, !events.isEmpty else {
    return
}
```

It's an optimization for `recordExperimentEvents`, but would make `notifyOnExperimentsApplied` less chatty.

Reply:
So we don't notify when experiments were applied if there wasn't any enrollment changes?  If so, I can just move the notify inside of the `if let events = events` block to accomplish the same thing.

Reply:
As for optimizing things, it might save pushing a function call onto the stack, but the `for in` loop shouldn't have much overhead on an empty array.